### PR TITLE
qa/rgw: Added Barbican secure key storage

### PR DESCRIPTION
**Pull request by aclamk**
Added task to deploy barbican. fixed keystone, rgw and s3tests tasks to work with it.
Added suite/rgw/crypt that tests against Barbican. Possibly other qa tasks should also use barbican in future, mostly multisite tests.

Signed-off-by: Adam Kupczyk <akupczyk@redhat.com>

### CODE REVIEW

File: qa/suites/rgw/crypt/overrides.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+overrides:
+  ceph:
+    wait-for-scrub: false
```
Comment (rzarzynski) [inline qa/suites/rgw/crypt/overrides.yaml]: Do we want to have this setting in *production* runs?

File: qa/suites/rgw/crypt/clusters/fixed-2.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../../clusters/fixed-2.yaml
```
Comment (rzarzynski) [inline qa/suites/rgw/crypt/clusters/fixed-2.yaml]: Maybe `fixed-1.yaml` would be fine?

File: qa/tasks/barbican.py [hunk 3]
```diff
@@ -0,0 +1,538 @@
...
+        ctx.cluster.only(client).run(
+            args=[
+                'git', 'clone',
+                '-b', branch,
+                'https://github.com/openstack/barbican.git',
+                '{tdir}/barbican'.format(tdir=testdir),
+                ],
+            )
+        if sha1 is not None:
+            ctx.cluster.only(client).run(
```
Comment (rzarzynski) [inline qa/tasks/barbican.py]: I propose to use `run_in_barbican_dir` here.

File: qa/tasks/keystone.py [hunk 4]
```diff
@@ -359,7 +366,7 @@ def task(ctx, config):
         or isinstance(config, dict), \
         "task keystone only supports a list or dictionary for configuration"
 
-    if not ctx.tox:
+    if not hasattr(ctx, 'tox'):
```
Comment (rzarzynski) [inline qa/tasks/keystone.py]: Thanks!

File: qa/tasks/rgw.py [hunk 5]
```diff
@@ -85,6 +85,29 @@ def start_rgw(ctx, config, clients):
...
                 ])
 
+        barbican_role = client_config.get('use-barbican-role', None)
+        if barbican_role is not None:
+            if not hasattr(ctx, 'barbican'):
+                raise ConfigError('rgw must run after the barbican task')
+
+            barbican_host, barbican_port = \
+                ctx.barbican.endpoints[barbican_role]
+            log.info("Use barbican url=%s:%s", barbican_host, barbican_port)
```
Comment (rzarzynski) [inline qa/tasks/rgw.py]: The log lacks the protocol part, and thus shouldn't mention *URL*. I guess we might want to assemble `barbican_url` first and then use it here and for `rgw_cmd` as well.

**Comment by mattbenjamin** [comment]
@cbodley I thought we merged all of the barbican work--is this still outstanding?

**Comment by cbodley** [comment]
@mattbenjamin this was the teuthology coverage, which i had forgotten about. would love to find someone to pick this up

**Comment by mattbenjamin** [comment]
@cb ok, will look for someone to help absorb this

**Comment by cbodley** [comment]
thanks to @aclamk @rzarzynski for doing the hard work here! this finally merged in https://github.com/ceph/ceph/pull/30218

**Pull request closed without merging**